src/path.js

```diff
@@ -16,6 +16,285 @@ function Path() {
+                }
+                break;
+            case 'C':
+                this.bezierCurveTo(...parsedBuffer);
+                break;
+            case 'Q':
+                this.quadraticCurveTo(...parsedBuffer);
+                break;
+            case 'Z':
+                if (this.commands.length < 1 || this.commands[this.commands.length - 1].type !== 'Z') {
+                    this.close();
+                }
+                break;
+        }
+
+        if (this.commands.length) {
+            for (const prop in this.commands[this.commands.length - 1]) {
+                if (this.commands[this.commands.length - 1][prop] === undefined) {
+                    this.commands[this.commands.length - 1][prop] = 0;
+                }
+            }
+        }
+    }
+
+    for (let i = 0; i < pathData.length; i++) {
+        const token = pathData.charAt(i);
+        const lastBuffer = buffer[buffer.length - 1];
+        if (number.indexOf(token) > -1) {
+            buffer[buffer.length - 1] += token;
+        } else if (sign.indexOf(token) > -1) {
+            if (!command.type && !this.commands.length) {
+                command.type = 'L';
+            }
+
+            if (token === '-') {
+                if (!command.type || lastBuffer.indexOf('-') > 0) {
+                    isUnexpected = true;
+                } else if (lastBuffer.length) {
+                    buffer.push('-');
+                } else {
+                    buffer[buffer.length - 1] = token;
+                }
+            } else {
+                if (!command.type || lastBuffer.length > 0) {
+                    isUnexpected = true;
+                } else {
+                    continue;
+                }
+            }
+        } else if (supportedCommands.indexOf(token) > -1) {
+            if (command.type) {
+                applyCommand.apply(this);
+                command = { type: token };
+            } else {
+                command.type = token;
+            }
+        } else if (unsupportedCommands.indexOf(token) > -1) {
+            // TODO: try to interpolate commands not directly supported?
+            throw new Error('Unsupported path command: ' + token + '. Currently supported commands are ' + supportedCommands.split('').join(', ') + '.');
+        } else if (' ,\t\n\r\f\v'.indexOf(token) > -1) {
+            buffer.push('');
+        } else if (token === '.') {
+            if (!command.type || lastBuffer.indexOf(token) > -1) {
+                isUnexpected = true;
+            } else {
+                buffer[buffer.length - 1] += token;
+            }
+        } else {
+            isUnexpected = true;
+        }
+
+        if (isUnexpected) {
+            throw new Error('Unexpected character: ' + token + ' at offset ' + i);
+        }
+    }
+    applyCommand.apply(this);
+
+    if (options.optimize) {
+        this.commands = optimizeCommands(this.commands);
+    }
+
+    let flipY = options.flipY;
```

Comment:
is this variable a number or a boolean ?

Reply:
See options documentation: "whether to flip the Y axis of the path data, because SVG and font paths use inverted Y axes. (true: calculate from bounding box, false: disable, integer/float: a base value for the calculation. default: true)"

It can be useful to calculate the actual base value from the font's ascender and descender values or the viewbox of an SVG you want to add the generated path to.

Reply:
I was referring to the left side `let flipY`

Reply:
It just takes on the value of the corresponding option. I can rewrite it if that's considered bad practice?

Reply:
I feel a litle bit confused when I see
```js
let flipY = options.flipY // boolean
```
then
```js
flipY = boundingBox.y1 + boundingBox.y2 // number ?!
```
but
```js
flipY !== false // boolean
```
and
```js
flipY - cmd[prop] // number again
```
I think there is 2 usecase for the same variable
the "offset" (number) version would be better in a separate variable I think.

Reply:
Makes sense that it might be confusing. Changed it with a new commit.

Reply:
Woops, pushed to the wrong repo, but now it's in.

Reply:
I still don't get why does we mixup in the same option/variable
- the offset
- the enabling
I would have used `{flipY:false}` `{flipY:true, flipOffsetY: 0.2}` if I get your algorithm correctly.

Reply:
So we *are* talking about the option itself, not just the variable?
I think it's just simpler to have one option for this. Either you disable it, have it calculated automatically, or you provide a value. You would never set `{flipY: false, flipOffsetY: 123}`, so in my mind it didn't make sense to have this constellation.
@ILOVEPIE what do you think? In the end, it doesn't really matter to me, so I'll go with the majority.

Reply:
split into `flipY` and `flipYBase` now, upated README accordingly